### PR TITLE
refactor(core): Rewrite cache on Dexie

### PR DESCRIPTION
**Pull request by ya-woodcutter**

### CODE REVIEW

**Comment by ya-woodcutter** [comment]
@TimeRaider @evilj0e @Blackheart340

File: src/index.js [hunk 1]
```diff
@@ -1,76 +1,164 @@
...
         }
 
-        result[storeName] = localforage.createInstance({
-            storeName,
-            name,
-            description,
-            driver
-        });
-
+        result[storeName] = '&';
```
Comment (TimeRaider) [inline src/index.js:23]: why?

**Comment by TimeRaider** [comment]
please add test coverage

File: src/index.js (hunk 1 above)
Comment (ya-woodcutter) [reply src/index.js:23]: For each store we need pairs `[key, value]` (field `key` - without name). `&` means that keys must be unique.

Comment (TimeRaider) [reply src/index.js:23]: i still don't get it, could you give me an example?
it's just that i don't really see the point of a hash table containing amps

**Comment by TimeRaider** [comment]
if you guys need to release that ASAP, i suggest you make a seperate branch named `next` and publish npm module with `@next` tag. [This article ](https://blog.greenkeeper.io/one-simple-trick-for-javascript-package-maintainers-to-avoid-breaking-their-user-s-software-and-to-6edf06dc5617#.vpvrscc5d) explains why it's a good idea to publish pre-lease code under @next tag

File: src/index.js (hunk 1 above)
Comment (ya-woodcutter) [reply src/index.js:23]: In Dexie stores initing looks like: 
``` 
db.version(version).stores({
    store1: "++field1, field2, field3",
    store2: "&field1, field2, field3",
});
```
`++` - Auto-incremented primary key
`&` - Unique index

In old cache we had one field - `key`, without name (this field must be unique).
Then I chose: 
```
{
    store1: '&' // Unique + without name
}
```

Comment (TimeRaider) [reply src/index.js:23]: 👍

**Comment by Blackheart340** [comment]
LGTM 👍

**Comment by TimeRaider** [comment]
I dug deeper into Dexie's docs and the lib looks very powerful, so I'm all for switching to it instead of localforage.
Since unity-cache's api hasn't changed, do we bump the minor version only?
Are there any shortcomings in current api, maybe i should be reworked as well? in that case we'll bump the major version.
Tell me what you think
... and awesome job, btw  👍 😄

**Comment by ya-woodcutter** [comment]
`... Since unity-cache's api hasn't changed ...`
Not exactly... we changed `cacheCreate` function (-description, +version).

I think we need bump the major version. And the saying goes, go majornaya versia ya sozdal ... but first, I need to fix coverage.

**Comment by TimeRaider** [comment]
@ya-woodcutter ok then, cool

**Comment by ya-woodcutter** [comment]
REVIEW: @TimeRaider @Blackheart340 @evilj0e 
I pushed the commit in upstream/rewrite-dexie :)

**Comment by TimeRaider** [comment]
@ya-woodcutter thanks! I'll play around with it this weekend

**Comment by Blackheart340** [comment]
@TimeRaider Looks like you forgot about it.
We discussed and thought it would be perfect if we'll get new version. So... don't be shy, just merge and release it :)

**Pull request merged**